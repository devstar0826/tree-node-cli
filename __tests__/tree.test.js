const fs = require('fs');
const junk = require('junk');
const path = require('path');
const tree = require('../tree');

const PATH_TO_TEST = '__tests__/fixtures';
const dirs = fs.readdirSync(PATH_TO_TEST).filter(junk.not);

describe('tree', () => {
  test('default', () => {
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir))).toMatchSnapshot();
    });
  });

  test('directoriesOnly', () => {
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        directoriesOnly: true,
      })).toMatchSnapshot();
    });
  });

  test('directoriesTrailingSlash', () => {
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        directoriesTrailingSlash: true,
      })).toMatchSnapshot();
    });
  });

  test('excludeDirs', () => {
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        excludeDirs: ['beta'],
      })).toMatchSnapshot();
    });
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        excludeDirs: ['charlie'],
      })).toMatchSnapshot();
    });
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        excludeDirs: ['beta', 'charlie'],
      })).toMatchSnapshot();
    });
  });

  test('hideHiddenFiles', () => {
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        hideHiddenFiles: false,
      })).toMatchSnapshot();
    });
  });

  test('maxDepth', () => {
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        maxDepth: 1,
      })).toMatchSnapshot();
    });
    dirs.forEach(dir => {
      expect(tree(path.join(PATH_TO_TEST, dir), {
        maxDepth: 2,
      })).toMatchSnapshot();
    });
  });
});
