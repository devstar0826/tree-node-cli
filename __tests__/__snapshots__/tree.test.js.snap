// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tree allFiles 1`] = `
"chain
└── alpha
    └── alpha
        └── alpha
            └── alpha
                └── alpha
                    └── alpha
                        └── alpha
                            └── alpha
                                └── alpha.txt"
`;

exports[`tree allFiles 2`] = `
"double-level
├── alpha.txt
└── beta
    └── alpha.txt"
`;

exports[`tree allFiles 3`] = `
"empty-dir
└── .gitkeep"
`;

exports[`tree allFiles 4`] = `
"multi-files
├── alpha.txt
├── beta.txt
└── charlie.txt"
`;

exports[`tree allFiles 5`] = `
"multi-level
├── alpha.txt
├── beta
│   ├── alpha
│   │   ├── alpha.txt
│   │   └── beta
│   │       ├── .gitkeep
│   │       └── alpha.txt
│   ├── beta.txt
│   └── charlie
│       └── alpha.txt
├── charlie
│   ├── .gitkeep
│   ├── alpha.txt
│   ├── beta
│   │   └── alpha.txt
│   └── charlie.txt
└── delta.txt"
`;

exports[`tree allFiles 6`] = `
"single-file
└── alpha.txt"
`;

exports[`tree allFiles 7`] = `
"single-level
├── alpha.txt
└── beta
    └── .gitkeep"
`;

exports[`tree default 1`] = `
"chain
└── alpha
    └── alpha
        └── alpha
            └── alpha
                └── alpha
                    └── alpha
                        └── alpha
                            └── alpha
                                └── alpha.txt"
`;

exports[`tree default 2`] = `
"double-level
├── alpha.txt
└── beta
    └── alpha.txt"
`;

exports[`tree default 3`] = `"empty-dir"`;

exports[`tree default 4`] = `
"multi-files
├── alpha.txt
├── beta.txt
└── charlie.txt"
`;

exports[`tree default 5`] = `
"multi-level
├── alpha.txt
├── beta
│   ├── alpha
│   │   ├── alpha.txt
│   │   └── beta
│   │       └── alpha.txt
│   ├── beta.txt
│   └── charlie
│       └── alpha.txt
├── charlie
│   ├── alpha.txt
│   ├── beta
│   │   └── alpha.txt
│   └── charlie.txt
└── delta.txt"
`;

exports[`tree default 6`] = `
"single-file
└── alpha.txt"
`;

exports[`tree default 7`] = `
"single-level
├── alpha.txt
└── beta"
`;

exports[`tree dirOnly 1`] = `
"chain
└── alpha
    └── alpha
        └── alpha
            └── alpha
                └── alpha
                    └── alpha
                        └── alpha
                            └── alpha"
`;

exports[`tree dirOnly 2`] = `
"double-level
└── beta"
`;

exports[`tree dirOnly 3`] = `"empty-dir"`;

exports[`tree dirOnly 4`] = `"multi-files"`;

exports[`tree dirOnly 5`] = `
"multi-level
├── beta
│   ├── alpha
│   │   └── beta
│   └── charlie
└── charlie
    └── beta"
`;

exports[`tree dirOnly 6`] = `"single-file"`;

exports[`tree dirOnly 7`] = `
"single-level
└── beta"
`;

exports[`tree exclude 1`] = `
"chain
└── alpha
    └── alpha
        └── alpha
            └── alpha
                └── alpha
                    └── alpha
                        └── alpha
                            └── alpha
                                └── alpha.txt"
`;

exports[`tree exclude 2`] = `
"double-level
├── alpha.txt"
`;

exports[`tree exclude 3`] = `"empty-dir"`;

exports[`tree exclude 4`] = `
"multi-files
├── alpha.txt
└── charlie.txt"
`;

exports[`tree exclude 5`] = `
"multi-level
├── alpha.txt
├── charlie
│   ├── alpha.txt
│   └── charlie.txt
└── delta.txt"
`;

exports[`tree exclude 6`] = `
"single-file
└── alpha.txt"
`;

exports[`tree exclude 7`] = `
"single-level
├── alpha.txt"
`;

exports[`tree exclude 8`] = `
"chain
└── alpha
    └── alpha
        └── alpha
            └── alpha
                └── alpha
                    └── alpha
                        └── alpha
                            └── alpha
                                └── alpha.txt"
`;

exports[`tree exclude 9`] = `
"double-level
├── alpha.txt
└── beta
    └── alpha.txt"
`;

exports[`tree exclude 10`] = `"empty-dir"`;

exports[`tree exclude 11`] = `
"multi-files
├── alpha.txt
├── beta.txt"
`;

exports[`tree exclude 12`] = `
"multi-level
├── alpha.txt
├── beta
│   ├── alpha
│   │   ├── alpha.txt
│   │   └── beta
│   │       └── alpha.txt
│   ├── beta.txt
└── delta.txt"
`;

exports[`tree exclude 13`] = `
"single-file
└── alpha.txt"
`;

exports[`tree exclude 14`] = `
"single-level
├── alpha.txt
└── beta"
`;

exports[`tree exclude 15`] = `
"chain
└── alpha
    └── alpha
        └── alpha
            └── alpha
                └── alpha
                    └── alpha
                        └── alpha
                            └── alpha
                                └── alpha.txt"
`;

exports[`tree exclude 16`] = `
"double-level
├── alpha.txt"
`;

exports[`tree exclude 17`] = `"empty-dir"`;

exports[`tree exclude 18`] = `
"multi-files
├── alpha.txt"
`;

exports[`tree exclude 19`] = `
"multi-level
├── alpha.txt
└── delta.txt"
`;

exports[`tree exclude 20`] = `
"single-file
└── alpha.txt"
`;

exports[`tree exclude 21`] = `
"single-level
├── alpha.txt"
`;

exports[`tree maxDepth 1`] = `
"chain
└── alpha"
`;

exports[`tree maxDepth 2`] = `
"double-level
├── alpha.txt
└── beta"
`;

exports[`tree maxDepth 3`] = `"empty-dir"`;

exports[`tree maxDepth 4`] = `
"multi-files
├── alpha.txt
├── beta.txt
└── charlie.txt"
`;

exports[`tree maxDepth 5`] = `
"multi-level
├── alpha.txt
├── beta
├── charlie
└── delta.txt"
`;

exports[`tree maxDepth 6`] = `
"single-file
└── alpha.txt"
`;

exports[`tree maxDepth 7`] = `
"single-level
├── alpha.txt
└── beta"
`;

exports[`tree maxDepth 8`] = `
"chain
└── alpha
    └── alpha"
`;

exports[`tree maxDepth 9`] = `
"double-level
├── alpha.txt
└── beta
    └── alpha.txt"
`;

exports[`tree maxDepth 10`] = `"empty-dir"`;

exports[`tree maxDepth 11`] = `
"multi-files
├── alpha.txt
├── beta.txt
└── charlie.txt"
`;

exports[`tree maxDepth 12`] = `
"multi-level
├── alpha.txt
├── beta
│   ├── alpha
│   ├── beta.txt
│   └── charlie
├── charlie
│   ├── alpha.txt
│   ├── beta
│   └── charlie.txt
└── delta.txt"
`;

exports[`tree maxDepth 13`] = `
"single-file
└── alpha.txt"
`;

exports[`tree maxDepth 14`] = `
"single-level
├── alpha.txt
└── beta"
`;

exports[`tree trailingSlash 1`] = `
"chain/
└── alpha/
    └── alpha/
        └── alpha/
            └── alpha/
                └── alpha/
                    └── alpha/
                        └── alpha/
                            └── alpha/
                                └── alpha.txt"
`;

exports[`tree trailingSlash 2`] = `
"double-level/
├── alpha.txt
└── beta/
    └── alpha.txt"
`;

exports[`tree trailingSlash 3`] = `"empty-dir/"`;

exports[`tree trailingSlash 4`] = `
"multi-files/
├── alpha.txt
├── beta.txt
└── charlie.txt"
`;

exports[`tree trailingSlash 5`] = `
"multi-level/
├── alpha.txt
├── beta/
│   ├── alpha/
│   │   ├── alpha.txt
│   │   └── beta/
│   │       └── alpha.txt
│   ├── beta.txt
│   └── charlie/
│       └── alpha.txt
├── charlie/
│   ├── alpha.txt
│   ├── beta/
│   │   └── alpha.txt
│   └── charlie.txt
└── delta.txt"
`;

exports[`tree trailingSlash 6`] = `
"single-file/
└── alpha.txt"
`;

exports[`tree trailingSlash 7`] = `
"single-level/
├── alpha.txt
└── beta/"
`;
